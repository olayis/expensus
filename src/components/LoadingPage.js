import React from 'react';
import loadingGif from '../../public/img/loading';

const LoadingPage = () => (
  <div className='loader'>
    <div>
      <img src={loadingGif} className='loader__image' />
      <p className='loader__text'>Loading...</p>
    </div>
  </div>
);

export default LoadingPage;
