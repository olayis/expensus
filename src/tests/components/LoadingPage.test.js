import React from 'react';
import { shallow } from 'enzyme';
import LoadingPage from '../../components/LoadingPage';
import splash from '../../../public/img/splash.gif';

test('should render LoadingPage correctly', () => {
  const wrapper = shallow(<LoadingPage />);
  expect(wrapper).toMatchSnapshot();
});

test('should render LoadingPage correctly with props', () => {
  const wrapper = shallow(
    <LoadingPage img={splash} width='262' height='98' hideLoadingText={true} />
  );
  expect(wrapper).toMatchSnapshot();
});
