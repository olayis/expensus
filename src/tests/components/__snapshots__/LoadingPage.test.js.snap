// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoadingPage correctly 1`] = `
<div
  className="loader"
>
  <div
    className="loader__content"
  >
    <img
      alt=""
      className="loader__image"
      height="200"
      src="test-file-stub"
      width="200"
    />
    <p
      className="loader__text"
    >
      Loading...
    </p>
  </div>
</div>
`;

exports[`should render LoadingPage correctly with props 1`] = `
<div
  className="loader"
>
  <div
    className="loader__content"
  >
    <img
      alt=""
      className="loader__image"
      height="98"
      src="test-file-stub"
      width="262"
    />
  </div>
</div>
`;
